Client won't launch on Linux due to dependency errors

<!--- Please follow the template below. Reports that do not follow the following template will be closed. -->

**Describe the bug**
Due to MonoGame removing the x86 specific dependencies of openal and sdl being removed the client is crashing when trying to export them.
 - Resources/x64/libopenal.so.1
 - Resources/x64/libSDL2-2.0.so.0

**Version Encountered**
0.6.2.432

**To Reproduce**
Try running the client on Linux

**Expected behavior**
The client to launch.

**Screenshots**
NA

**Logs**
Unhandled Exception:
System.ArgumentNullException: Value cannot be null.
Parameter name: stream
  at System.Resources.ResourceReader..ctor (System.IO.Stream stream) [0x00009] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.Resources.ResourceSet..ctor (System.IO.Stream stream) [0x00006] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at Intersect.Client.Program.ExportDependency (System.String filename, System.String folder, System.String nameoverride) [0x00121] in <b0091a5914e24f6093b88ba56f8c7326>:0 
  at Intersect.Client.Program.ExportDependencies () [0x000f2] in <b0091a5914e24f6093b88ba56f8c7326>:0 
  at Intersect.Client.Program.Main () [0x00017] in <b0091a5914e24f6093b88ba56f8c7326>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentNullException: Value cannot be null.
Parameter name: stream
  at System.Resources.ResourceReader..ctor (System.IO.Stream stream) [0x00009] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at System.Resources.ResourceSet..ctor (System.IO.Stream stream) [0x00006] in <9f0df102fe6e4cfea29d2e46f585d8a5>:0 
  at Intersect.Client.Program.ExportDependency (System.String filename, System.String folder, System.String nameoverride) [0x00121] in <b0091a5914e24f6093b88ba56f8c7326>:0 
  at Intersect.Client.Program.ExportDependencies () [0x000f2] in <b0091a5914e24f6093b88ba56f8c7326>:0 
  at Intersect.Client.Program.Main () [0x00017] in <b0091a5914e24f6093b88ba56f8c7326>:0

**Operating System:**
Linux

**Additional context**
NA

--- a/Intersect.Client/Intersect.Client.csproj
+++ b/Intersect.Client/Intersect.Client.csproj
@@ -233,18 +233,12 @@
     </EmbeddedResource>
     <None Include="app.config" />
     <None Include="packages.config" />
-    <EmbeddedResource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libSDL2-2.0.so.0">
-      <Link>Resources\x86\libSDL2</Link>
-    </EmbeddedResource>
-    <EmbeddedResource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libSDL2-2.0.so.0">
-      <Link>Resources\x64\libSDL2</Link>
-    </EmbeddedResource>
-    <EmbeddedResource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libopenal.so.1">
-      <Link>Resources\x86\libopenal</Link>
-    </EmbeddedResource>
-    <EmbeddedResource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libopenal.so.1">
-      <Link>Resources\x64\libopenal</Link>
-    </EmbeddedResource>
+    <Resource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libSDL2-2.0.so.0">
+      <Link>Resources\x64\libSDL2-2.0.so.0</Link>
+    </Resource>
+    <Resource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libopenal.so.1">
+      <Link>Resources\x64\libopenal.so.1</Link>
+    </Resource>
     <EmbeddedResource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\osx\native\libSDL2-2.0.0.dylib">
       <Link>Resources\libSDL2-2.0.0.dylib</Link>
     </EmbeddedResource>

--- a/Intersect.Client/Intersect.Client.csproj
+++ b/Intersect.Client/Intersect.Client.csproj
@@ -234,10 +234,10 @@
     <None Include="app.config" />
     <None Include="packages.config" />
     <Resource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libSDL2-2.0.so.0">
-      <Link>Resources\x64\libSDL2-2.0.so.0</Link>
+      <Link>Resources\x64\libSDL2</Link>
     </Resource>
     <Resource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\linux-x64\native\libopenal.so.1">
-      <Link>Resources\x64\libopenal.so.1</Link>
+      <Link>Resources\x64\libopenal</Link>
     </Resource>
     <EmbeddedResource Include="..\packages\MonoGame.Framework.DesktopGL.3.8.0.1641\runtimes\osx\native\libSDL2-2.0.0.dylib">
       <Link>Resources\libSDL2-2.0.0.dylib</Link>

--- a/Intersect.Client/MonoGame/File Management/MonoContentManager.cs
+++ b/Intersect.Client/MonoGame/File Management/MonoContentManager.cs
@@ -38,12 +38,12 @@ namespace Intersect.Client.MonoGame.File_Management
             {
                 var realFilename = tilesetFiles.FirstOrDefault(file => t.Equals(file, StringComparison.InvariantCultureIgnoreCase));
                 if (t != "" &&
-                    (realFilename != null ||
+                    (!string.IsNullOrWhiteSpace(realFilename) ||
                      GameTexturePacks.GetFrame(Path.Combine("resources", "tilesets", t.ToLower())) != null) &&
                     !mTilesetDict.ContainsKey(t.ToLower()))
                 {
                     mTilesetDict.Add(
-                        t.ToLower(), Core.Graphics.Renderer.LoadTexture(Path.Combine("resources", "tilesets", realFilename != null ? realFilename : t))
+                        t.ToLower(), Core.Graphics.Renderer.LoadTexture(Path.Combine("resources", "tilesets", !string.IsNullOrWhiteSpace(realFilename) ? realFilename : t))
                     );
                 }
             }

--- a/Intersect.Client/MonoGame/File Management/MonoContentManager.cs
+++ b/Intersect.Client/MonoGame/File Management/MonoContentManager.cs
@@ -1,7 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
-
+using System.Linq;
 using Intersect.Client.Framework.File_Management;
 using Intersect.Client.Framework.GenericClasses;
 using Intersect.Client.Framework.Graphics;
@@ -33,15 +33,17 @@ namespace Intersect.Client.MonoGame.File_Management
         public override void LoadTilesets(string[] tilesetnames)
         {
             mTilesetDict.Clear();
+            var tilesetFiles = Directory.GetFiles(Path.Combine("resources", "tilesets")).Select(f => Path.GetFileName(f));
             foreach (var t in tilesetnames)
             {
+                var realFilename = tilesetFiles.FirstOrDefault(file => t.Equals(file, StringComparison.InvariantCultureIgnoreCase));
                 if (t != "" &&
-                    (File.Exists(Path.Combine("resources", "tilesets", t)) ||
+                    (realFilename != null ||
                      GameTexturePacks.GetFrame(Path.Combine("resources", "tilesets", t.ToLower())) != null) &&
                     !mTilesetDict.ContainsKey(t.ToLower()))
                 {
                     mTilesetDict.Add(
-                        t.ToLower(), Core.Graphics.Renderer.LoadTexture(Path.Combine("resources", "tilesets", t))
+                        t.ToLower(), Core.Graphics.Renderer.LoadTexture(Path.Combine("resources", "tilesets", realFilename != null ? realFilename : t))
                     );
                 }
             }

--- a/Intersect.Client/Program.cs
+++ b/Intersect.Client/Program.cs
@@ -142,6 +142,8 @@ namespace Intersect.Client
                 }
             }
 
+            platformId = PlatformID.Unix;
+
             var folder = Environment.Is64BitProcess ? "x64" : "x86";
 
             switch (platformId)
@@ -168,8 +170,8 @@ namespace Intersect.Client
 
                 case PlatformID.Unix:
                 default:
-                    ExportDependency("libopenal.so.1", folder);
-                    ExportDependency("libSDL2-2.0.so.0", folder);
+                    ExportDependency("libopenal.so.1", "x64");
+                    ExportDependency("libSDL2-2.0.so.0", "x64");
                     ExportDependency("openal32.dll", "");
                     ExportDependency("MonoGame.Framework.dll.config", "", "MonoGame.Framework.Client.dll.config");
 

--- a/Intersect.Client/Program.cs
+++ b/Intersect.Client/Program.cs
@@ -142,8 +142,6 @@ namespace Intersect.Client
                 }
             }
 
-            platformId = PlatformID.Unix;
-
             var folder = Environment.Is64BitProcess ? "x64" : "x86";
 
             switch (platformId)

